### DISCUSSION TITLE
org-roam ids instead of links

### DISCUSSION THREAD

**Discussion by damionx7** [damionx7/Zettel-Notes-Documentation#159] [Feature Requests]
[Daniel Thom]:

This will make Zettel notes THE APP for org-mode and org-roam users

**Comment by damionx7** [comment]
org-roam works with ids and a little Database in the background.
Every org-roam file has an unique id.

This is a sample header

```
:PROPERTIES:
:ID:       4A2217C7-3C51-46A7-8443-8DDBD542D85F
:END:
#+title: 2024-09-29
```
So i think a solution could be a table with this id and a mapping to a path in the file system

Note Link: 
`[[id:2be41782-935e-4cf8-8986-9b4a7deb1260][Title]]`

**Comment by dvictori** [comment]
I just installed Zettel notes for viewing my org-roam notes on my phone. The app looks great and works fine except, I don't know if it's possible to use the org-roam ID links to go from one note to another. Is this feature available?

I saw that v 2.9.9 included support for parsing org-roam ids. Does that include following the links and identifying backlinks?

**Reply by damionx7** [reply]
Yes. Its possible to use org-roam ids to navigate and create links between files.

**Reply by dvictori** [reply]
Is there some special settings to use? I just loaded my org-roam files. The links are rendered but when I try to follow I get a note link not valid error.

In EMACS the links works fine

**Reply by damionx7** [reply]
Can you share the note file

**Reply by dvictori** [reply]
I can´t attach org files (github does not like it). So I'm pasting two file contents here that I just tried and id did not work

file 1: 20260311130902-test_zettel_parent.org
------------------------------------
```
:PROPERTIES:
:ID:       fbedc776-df33-4236-b986-97403a09c871
:END:
#+title: test_zettel_parent
#+date: [2026-03-11 Wed 15:09]

This is a parent note for testing Zettel Notes

This is an URL link for [[https://www.zettelnotes.com/][Zettel Notes]]
This link works fine on Zettel Notes on Android and opens the webpage 

This is a link to another org-roam file that uses org-roam ids:  [[id:fb640e4f-e94a-41c3-ad8b-d20f60e3998f][test_zettel_child]]
This link gives a *Note link not valid* error.
```
-----------------------------------------------

file 2: 20260311161101-test_zettel_child.org
-----------------------------------------------------
```
:PROPERTIES:
:ID:       fb640e4f-e94a-41c3-ad8b-d20f60e3998f
:END:
#+title: test_zettel_child
#+date: [2026-03-11 Wed 15:11]

this is a child note from [[id:fbedc776-df33-4236-b986-97403a09c871][test_zettel_parent]]
```
-----------------------------------

**Reply by damionx7** [reply]
https://github.com/user-attachments/assets/c51f1003-ba42-4b62-b17d-1cf82bf5a4e1

These id links are working. Pls check the video.

**Reply by dvictori** [reply]
Many thanks for looking at this. Unfortunately for me it's not working. Could it be that it's because I created the org files on emacs org-roam and pulled it into Zettel using webdav? Or is there a specific config that I'm missing?

Below is a screen capture of what I get

https://github.com/user-attachments/assets/6b62dedf-f6a4-408b-98b6-443d30fa9323

**Reply by damionx7** [reply]
Can you try creating these two note files again.

**Reply by dvictori** [reply]
I created the two files inside the "App folder" repository. I found out that I cannot "click" on the links in the files. But if I open the note info panel (not sure how that's called), I can see the backlinks and go to linked note.

This does not work for the files that are in another repository which I pulled from webdav


https://github.com/user-attachments/assets/ecd39850-c187-48a2-a1d1-14571ae470de

**Reply by damionx7** [reply]
What app version are you using? Also can you share log file from Zettel Notes > Settings > Misc > Logs.

**Reply by dvictori** [reply]
I'm using v3.0.3 from Google Play Store.

The log file is attached. There are some connection errors but those were caused by my faulty internet connection

[zettel_app_20260316.log](https://github.com/user-attachments/files/26036044/zettel_app_20260316.log)

**Reply by damionx7** [reply]
Can you try creating a repository on internal storage and try the links

**Reply by dvictori** [reply]
Sorry I missed your message from mar22. I did not understand what you asked. The repository I'm testing is inside the `App Folder` (which I think it's internal storage). What I've seen is that on those files inside `App Folder`, when I click on a link id does not work. But the links are shown and work on the "note detail" menu

**Comment by irmod** [comment]
@dvictori, special applications exist and are being actively developed to work with the org-mode format:
 - Logseq,
 - Orgzly,
 - Orgzly Revived (fork from Orgzly),
 - Orgro

**Comment by dvictori** [comment]
Thanks. I use Orgzly Revived for projects and tasks. But for org-roam notes, since the amount of org files tends to be very large, it becomes impractical.

I heard about logseq and I need to check it out.
